packages/compiler-cli/src/ngtsc/transform/src/compilation.ts

```diff
@@ -259,19 +257,25 @@ export class TraitCompiler implements ProgramTypeCheckAdapter {
     let record: ClassRecord|null = this.recordFor(clazz);
     let foundTraits: PendingTrait<unknown, unknown, SemanticSymbol|null, unknown>[] = [];
 
-    // A set to track the non-Angular decorators in local compilation mode. An error will be issued
-    // if non-Angular decorators is found in local compilation mode.
-    const nonNgDecoratorsInLocalMode =
-        this.compilationMode === CompilationMode.LOCAL ? new Set(decorators) : null;
+    // A set to track and forbid non-Angular decorators. Such ban is needed in some cases where
+    // custom decorators are not supported (yet). Current cases are: 1) local compilation mode, 2)
+    // when option `forbidOrphanComponents` is set. These two cases share the same solution as the
+    // reason for both cases is that the `DepsTracker` is not able to understand components mutated
+    // by decorators. Once the `DepsTracker` is fixed then these issues will be resolved. This issue
+    // is to be fixed soon (b/320536434).
+    const unmatchedDecoratorsInLocalMode =
```

Comment:
```suggestion
    const unmatchedDecorators =
```

Regarding "fixing deps tracker". There is no fix we are landing right? except hoping for ES decorators at some point. The comment sound like deps tracker needs a fix.

Reply:
I was just curious because of the comment sounding like we intend to allow this in the future again. Maybe remove?